### PR TITLE
More control barrier tests

### PR DESCRIPTION
**Pull request by hernanponcedeleon**
@tonghaining can you take a look?

### CODE REVIEW

File: dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java [hunk 1]
```diff
@@ -961,7 +961,9 @@ public Knowledge visitSyncBarrier(Relation sync_bar) {
             List<FenceWithId> fenceEvents = program.getThreadEvents(FenceWithId.class);
             for (FenceWithId e1 : fenceEvents) {
                 for (FenceWithId e2 : fenceEvents) {
-                    if(exec.areMutuallyExclusive(e1, e2)) {
+                    // “A bar.sync or bar.red or bar.arrive operation synchronizes with a bar.sync
```
Comment (tonghaining) [inline dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java:964]: what is `bar.red`?

Comment (tonghaining) [reply dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java:964]: Okay, I think the current implementation support (`bar.sync`, `bar.sync`) and (`bar.arrive`, `bar.sync`).
Shall we also update the comments on two ptx cat files about the `sync_barrier` relation?

Comment (hernanponcedeleon) [reply dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java:964]: I don't think we need that level of detail in the cat file. E.g. we do not specify either all the cases covered by `sr`.

**Comment by tonghaining** [comment]
LGTM

**Pull request merged**